module/modmaninterface/interface.go

```diff
@@ -0,0 +1,23 @@
+// Package modmaninterface abstracts the manager interface to avoid an import cycle/loop.
+package modmaninterface
+
+import (
+	"context"
+
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/resource"
+)
+
+// ModuleManager abstracts the module manager interface.
+type ModuleManager interface {
+	AddModule(ctx context.Context, cfg config.Module) error
+
+	AddResource(ctx context.Context, cfg config.Component, deps []string) (interface{}, error)
+	ReconfigureResource(ctx context.Context, cfg config.Component, deps []string) error
+	RemoveResource(ctx context.Context, name resource.Name) error
+	IsModularResource(name resource.Name) bool
```

Comment:
It feels like as a user `IsModularResource` and `NeedsModule` are basically the same. Can we just merge them and expect a resource.Name (or better Type) to always be passed?